client/src/components/generic/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,29 @@
+type checkboxVariants = "default" | "checked" | "disabled"
```

Comment:
nit: disabled should already be part of `React.InputHTMLAttributes<HTMLInputElement>`